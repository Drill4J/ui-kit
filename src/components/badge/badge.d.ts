/// <reference types="react" />
interface Props {
    className?: string;
    children?: React.ReactNode;
    bold?: boolean;
    color?: 'green' | 'orange' | 'gray' | 'red';
}
export declare const Badge: ({ className, children }: Props) => JSX.Element;
export {};
