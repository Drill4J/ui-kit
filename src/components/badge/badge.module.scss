@import '../../common/style-guide/index';

.badge {
  border: 1px solid $slate-gray;
  border-radius: 14px;
  font-size: 11px;
  line-height: 12px;
  padding: 4px 10px;
}
