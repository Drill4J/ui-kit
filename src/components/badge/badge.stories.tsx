import React from 'react';
import { storiesOf } from '@storybook/react';

import { Badge } from './badge';

storiesOf('Badge', module)
  .add('with simple and bold text', () => (
    <div style={{ display: 'flex', justifyContent: 'space-between', margin: '40px 0px' }}>
      <Badge>simple</Badge>
      <Badge bold>bold</Badge>
      <Badge color="green">Online</Badge>
      <Badge color="orange">Busy</Badge>
      <Badge color="red">Offline</Badge>
    </div>
  ));
