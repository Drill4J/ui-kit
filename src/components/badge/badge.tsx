import styled, { css } from 'styled-components';

import { COLORS, FONTS } from '../../theme';

interface Props {
  className?: string;
  children?: React.ReactNode;
  bold?: boolean;
  color?: 'green' | 'orange' | 'gray' | 'red';
}

export const Badge = styled.span<Props>`
  padding: 0 8px;
  border: 1px solid currentColor;
  border-radius: 11px;
  font-size: 12px;
  line-height: 20px;
  ${({ bold }) => bold && css`font-family: ${FONTS.SEMI_BOLD}`}
  ${({ color }) => [
    color === 'gray' && css`
      border: 1px solid ${COLORS.MONOCHROME.DEFAULT};
      color: ${COLORS.MONOCHROME.DEFAULT};
    `,
    color === 'green' && css`
      border: 1px solid ${COLORS.GREEN.DEFAULT};
      color: ${COLORS.GREEN.DEFAULT};
    `,
    color === 'orange' && css`
      border: 1px solid ${COLORS.ORANGE.DEFAULT};
      color: ${COLORS.ORANGE.DEFAULT};
    `,
    color === 'red' && css`
      border: 1px solid ${COLORS.RED.DEFAULT};
      color: ${COLORS.RED.DEFAULT};
    `,
  ]}
`;
