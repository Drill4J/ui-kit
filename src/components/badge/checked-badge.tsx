import 'twin.macro';
import { Icons } from '../icon';

export const CheckedBadge = () => (
  <div tw="flex items-center justify-center w-3 h-3 rounded-full bg-green-default">
    <Icons.Check width={6} height={6} />
  </div>
);
