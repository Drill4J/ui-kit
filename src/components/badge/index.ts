export { Badge } from './badge';
export { CheckedBadge } from './checked-badge';
