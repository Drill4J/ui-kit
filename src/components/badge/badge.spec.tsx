import React from 'react';
import renderer from 'react-test-renderer';

import { Badge } from './badge';

describe('Badge', () => {
  it('should match snapshot with simple text', () => {
    const tree = renderer.create(<Badge color="green">Online</Badge>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with bold text', () => {
    const tree = renderer.create(<Badge color="green" bold>Online</Badge>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
