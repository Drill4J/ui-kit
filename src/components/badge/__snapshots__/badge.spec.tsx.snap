// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<span
  className="badge"
>
  simple
</span>
`;

exports[`should match snapshot with bold 1`] = `
<span
  className="badge badge--bold"
>
  bold
</span>
`;
